Conflict between --with-openssl and --with-jemalloc build options

While building with `--with-jemalloc`, I keep getting the error message `configure: error: jemalloc has bogus version`. After some digging I found that this started happening after commit https://github.com/apache/trafficserver/commit/3412c4f69552b10779cfe0ce1bc6a71971734491

After that commit, if I specify both `--with-openssl` and `--with-jemalloc` options, I get the error message, if I only specify `--with-jemalloc` everything works as expected.

--- a/build/crypto.m4
+++ b/build/crypto.m4
@@ -385,6 +385,7 @@ AC_DEFUN([TS_CHECK_SESSION_TICKET], [
 dnl SSL_set1_verify_cert_store macro is for OpenSSL 1.1.1
 dnl SSL_set1_verify_cert_store function is for BoringSSL
 AC_DEFUN([TS_CHECK_VERIFY_CERT_STORE], [
+  _saved_LIBS=$LIBS
 
   TS_ADDTO(LIBS, [$OPENSSL_LIBS])
   AC_CHECK_HEADERS(openssl/ssl.h)
@@ -415,6 +416,8 @@ AC_DEFUN([TS_CHECK_VERIFY_CERT_STORE], [
     ],
     []
   )
+
+  LIBS=$_saved_LIBS
 
   AC_MSG_CHECKING([for setting verify cert store APIs])
   AC_MSG_RESULT([$verify_cert_store_check])

